treeherder/seta/job_priorities.py

```diff
@@ -0,0 +1,142 @@
+import datetime
+import logging
+
+from treeherder.etl.seta import Treecodes
+from treeherder.model.models import Repository
+from treeherder.seta.common import unique_key
+from treeherder.seta.models import (JobPriority,
+                                    TaskRequest)
+
+GECKO_DECISION_TASK_USER_AGENT = 'TaskCluster'
+LOG = logging.getLogger(__name__)
+RESET_DELTA = 5400
+
+
+class SetaError(Exception):
+    pass
+
+
+# XXX: We can get rid of this function if Treecodes took care of it
+def _ref_data_names(build_system):
+    '''
+    Sample data from Treecodes().query_jobnames() (skipping irrelevant fields)
+    {
+        "buildplatform": "buildbot",
+        "buildtype": "debug",
+        "name": "web-platform-tests-1",
+        "platform": "windows8-64",
+        "ref_data_name": "Windows 8 64-bit mozilla-inbound debug test web-platform-tests-1",
+    },{
+        "buildplatform": "taskcluster",
+        "buildtype": "opt",
+        "name": "mochitest-webgl-e10s-1",
+        "platform": "linux64",
+        "ref_data_name": "desktop-test-linux64/opt-mochitest-webgl-e10s-1",
+    }
+    '''
+    ref_data_names = {}
+    for job in Treecodes().query_jobnames():
+        key = unique_key(testtype=job['name'],
+                         buildtype=job['buildtype'],
+                         platform=job['platform'])
+        if build_system == '*':
+            ref_data_names[key] = job['ref_data_name']
+        elif job['buildplatform'] == build_system:
+            ref_data_names[key] = job['ref_data_name']
+
+    return ref_data_names
+
+
+def _process(build_system, job_priorities):
+    '''Return list of ref_data_name for job_priorities'''
+    jobs = []
+    # This map contains the ref_data_name of every Treeherder job
+    ref_data_names = _ref_data_names(build_system)
+
+    for jp in job_priorities:
+        if build_system == 'taskcluster':
+            # We need to retranslate the jobtype back to the proper data form after all.
+            # Bug 1313844 - Name inconsistencies between Buildbot and TaskCluster
+            jp.testtype = jp.testtype.replace('e10s-browser-chrome', 'browser-chrome-e10s')
+            jp.testtype = jp.testtype.replace('e10s-devtools-chrome', 'devtools-chrome-e10s')
+            jp.testtype = jp.testtype.replace('gl-', 'webgl-')
```

Comment:
this code will be invalid or more useful in the land of BBB- I expect it to change in the coming weeks.